src/Zio/FileSystems/PhysicalFileSystem.cs

```diff
@@ -968,6 +968,11 @@ protected override string ConvertPathToInternalImpl(UPath path)
         return absolutePath;
     }
 
+    private static bool HasWindowsVolumeLabel(string path)
+    {
+        return path.IndexOf(":\\", StringComparison.Ordinal) == 1;
```

Comment:
Should we test for the Windows OS here as well? (e.g OSPlatform or equivalent)

Reply:
(Faster check would be `path.Length >= 3 && path[1] == ':' && path[2] == '\\'`)

Reply:
Done, thanks for the feedback.